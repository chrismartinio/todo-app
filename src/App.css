.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

.remove-button {
  margin-left: 10px;
  cursor: pointer;
}

.edit-button {
  margin-left: 10px;
}

.hidden {
  display: none;
}


