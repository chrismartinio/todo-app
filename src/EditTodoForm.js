import React, { Component } from 'react';

class EditTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = { task: this.props.task }
    this.handleChange = this.handleChange.bind(this);
    // this.handleSubmit = this.handleSubmit.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value })
  }

  // handleSubmit(e) {
  //   e.preventDefault();
  //   this.props.addTask(this.state);
  //   this.setState({ task: '' });
  // }

  handleEdit(e) {
    e.preventDefault();
    this.props.updateTask(this.props.id, this.state);
    this.props.hideForm(this.props.id);
  }



  render() {
    return (
      <div className="hidden" id={`${this.props.id}`}>

        <form onSubmit={this.handleEdit}>
   
          <input id="task" name="task" value={this.state.task} onChange={this.handleChange}></input>

        <button>Edit task</button>
        </form>

      </div>
    )
  }
}

export default EditTodoForm;