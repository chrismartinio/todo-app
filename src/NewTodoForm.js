import React, { Component } from 'react';

class NewTodoForm extends Component {
  constructor(props) {
    super(props);
    this.state = { task: '' }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {

    this.setState({ [e.target.name]: e.target.value })
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.addTask(this.state);
    this.setState({ task: '' });
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
        <label htmlFor="task">Enter task: </label>
        <input id="task" name="task" value={this.state.task} onChange={this.handleChange}></input>

        <button>Add task</button>

        </form>
      </div>
    )
  }
}


export default NewTodoForm

