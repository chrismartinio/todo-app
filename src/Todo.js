import React, { Component } from 'react';
import EditTodoForm from './EditTodoForm';

class Todo extends Component {

  constructor(props) {
    super(props);
    this.handleRemove = this.handleRemove.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
  }

  handleRemove(e) {
    e.preventDefault();
    this.props.removeTask(this.props.id);
  }

  handleEdit(e) {
    e.preventDefault();
    this.props.showForm(this.props.id);
  }

  render() {
    return (
      <div>
        <p>{this.props.task}
          <button className="edit-button" onClick={this.handleEdit}>edit</button>
          <button className="remove-button" onClick={this.handleRemove}>x</button>
        </p>
      </div>
    )
  }
}

export default Todo;

