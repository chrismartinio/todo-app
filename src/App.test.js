import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from "enzyme"
import toJson from "enzyme-to-json";
import { exportAllDeclaration } from '@babel/types';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import EditTodoForm from './EditTodoForm';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NewTodoForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Todo />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EditTodoForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", function() {
  let wrapper = shallow(<Todo />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot()
})

