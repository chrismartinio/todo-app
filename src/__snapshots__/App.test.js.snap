// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <p>
    <button
      className="edit-button"
      onClick={[Function]}
    >
      edit
    </button>
    <button
      className="remove-button"
      onClick={[Function]}
    >
      x
    </button>
  </p>
</div>
`;
